upgrades.go

```diff
@@ -0,0 +1,186 @@
+package syscfg
+
+// This file contains tweaks for enabling/disabling unattended upgrades.
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	errw "github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	autoUpgradesPath             = "/etc/apt/apt.conf.d/20auto-upgrades"
+	autoUpgradesContentsEnabled  = `APT::Periodic::Update-Package-Lists "1";` + "\n" + `APT::Periodic::Unattended-Upgrade "1";` + "\n"
+	autoUpgradesContentsDisabled = `APT::Periodic::Update-Package-Lists "1";` + "\n" + `APT::Periodic::Unattended-Upgrade "0";` + "\n"
+
+	unattendedUpgradesPath = "/etc/apt/apt.conf.d/50unattended-upgrades"
+)
+
+type UpgradesConfig struct {
+	// Type can be
+	// Empty/missing ("") to make no changes
+	// "disable" (or "disabled") to disable auto-upgrades
+	// "security" to enable ONLY security upgrades
+	// "all" to enable upgrades from all configured sources
+	Type string `json:"type"`
+}
+
+func EnforceUpgrades(ctx context.Context, cfg UpgradesConfig, log *zap.SugaredLogger) {
+	if cfg.Type == "" {
+		return
+	}
+
+	err := checkSupportedDistro()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	if cfg.Type == "disable" || cfg.Type == "disabled" {
+		isNew, err := writeFileIfNew(autoUpgradesPath, []byte(autoUpgradesContentsDisabled))
```

Comment:
out of scope for this review, but might want to add docstring to the `writeFileIfNew` function definition saying this writes file if the contents are different; 'new' is ambiguous

Reply:
Will note that for the big code merge for agent.